stonesoup/plotting.py

```diff
@@ -0,0 +1,282 @@
+                         **tracks_kwargs)
+
+        # Generate legend items for track
+        track_handle = Line2D([], [], linestyle=tracks_kwargs['linestyle'],
+                              marker=tracks_kwargs['marker'], color='black')
+        track_label = "Track"
+        self.handles_list.append(track_handle)
+        self.labels_list.append(track_label)
+
+        if uncertainty:
+            # Plot uncertainty ellipses
+            for track in tracks:
+                HH = np.eye(track.ndim)[mapping, :]  # Get position mapping matrix
+                for state in track:
+                    w, v = np.linalg.eig(HH @ state.covar @ HH.T)
+                    max_ind = np.argmax(w)
+                    min_ind = np.argmin(w)
+                    orient = np.arctan2(v[1, max_ind], v[0, max_ind])
+                    ellipse = Ellipse(xy=(state.state_vector[0], state.state_vector[2]),
+                                      width=2 * np.sqrt(w[max_ind]),
+                                      height=2 * np.sqrt(w[min_ind]),
+                                      angle=np.rad2deg(orient), alpha=0.2,
+                                      color=tracks_kwargs['color'])
+                    self.ax.add_artist(ellipse)
+
+            # Generate legend items for uncertainty ellipses
+            ellipse_handle = Ellipse((0.5, 0.5), 0.5, 0.5, alpha=0.2, color=tracks_kwargs['color'])
+            ellipse_label = "Uncertainty"
+
+            self.handles_list.append(ellipse_handle)
+            self.labels_list.append(ellipse_label)
+
+            # Generate legend
+            self.ax.legend(handles=self.handles_list, labels=self.labels_list,
+                           handler_map={Ellipse: _HandlerEllipse()})
+
+        elif particle:
+            # Plot particles
+            for track in tracks:
+                for state in track:
+                    data = np.array([particle.state_vector for particle in state.particles])
+                    self.ax.plot(data[:, 0], data[:, 2], linestyle='', marker=".",
+                                 markersize=1, alpha=0.5)
+
+            # Generate legend items for particles
+            particle_handle = Line2D([], [], linestyle='', color="black", marker='.',
+                                     markersize=1)
+            particle_label = "Particles"
+            self.handles_list.append(particle_handle)
+            self.labels_list.append(particle_label)
+
+            # Generate legend
+            self.ax.legend(handles=self.handles_list, labels=self.labels_list)
+
+        else:
+            self.ax.legend(handles=self.handles_list, labels=self.labels_list)
+
+    # Ellipse legend patch (used in Tutorial 3)
+    @staticmethod
+    def ellipse_legend(ax, label_list, color_list, **kwargs):
+        """Adds an ellipse patch to the legend on the axes. One patch added for each item in `label_list`
+        with the corresponding color from `color_list`.
```

Comment:
Limited to 100 chars per line
```suggestion
        """Adds an ellipse patch to the legend on the axes. One patch added for each item in 
        `label_list` with the corresponding color from `color_list`.
```